### DISCUSSION TITLE
can airflow support mermaid grammar?

### DISCUSSION THREAD

**Discussion by maijh** [apache/airflow#13335] [Q&A] [answered]
when i use airflow doc feature ， i found it not support markdown mermaid grammar，Can someone answer the question?

**Comment by potiuk** [comment, marked as answer]
It is not supported directly, but we have some mermaid files that are automatically converted into images by our pre-commits (see for example here https://github.com/apache/airflow/tree/master/images/ci) and then you can embed those generated images in the documentation.

**Reply by maijh** [reply]
Do you mean to use mermaid to generate the graph, and then use markdown to insert the graph？what's the usage of md5 file?

**Reply by potiuk** [reply]
Yes. MD5 is standard MD5 sum of the mermaid file is generated automatically when you generate image to check if the original .mermaid file has changed. 

The only thing you need to do is install `pre-commit` (`pre-commit install`) as described in  https://github.com/apache/airflow/blob/master/STATIC_CODE_CHECKS.rst#prerequisites-for-pre-commit-hooks and every time you make a change to the .mermaid file , the image will get regenerated automatically.

You can also do it by running `scripts/ci/pre_commit/pre_commit_mermaid.sh <your mermaid file>`. It converts the mermaid file to image if it changed (compares the file md5 sum with the stored one and will not regenerate the image if the mermaid file has not changed).

**Reply by maijh** [reply]
Sorry, I still don’t understand. When I generate the image, how do I import this ? Can you write a piece of code and describe how to import the image in markdown，thank you.

**Reply by potiuk** [reply]
See how it is done in CI.rst https://github.com/apache/airflow/blob/master/CI.rst#ci-sequence-diagrams
